### DISCUSSION TITLE
How to type `compose` function ?

### DISCUSSION THREAD

**Discussion by kaguya3222** [ramda/ramda#3346] [Q&A]
I want to use `compose` function with a wrapper function when I want to pass some functions that should be composed with another function. 

Example:

```
import { compose } from "ramda";

const foo = (a: number) => a;

const myCompose = (fns: Array<(a: number) => number>) => {
  return compose(...fns, foo);
};
``` 

But I get TS error: `A spread argument must either have a tuple type or be passed to a rest parameter.ts(2556)`

Demo: [Codesandbox](https://codesandbox.io/s/confident-forest-1vnnwz?file=/src/index.ts:0-160)

Is there some solution for this?

**Comment by Harris-Miller** [comment]
The error you are getting is not from Ramda or how `composed` is defined, but rather typescript itself. It's saying that you cannot spread `fns` and then add `foo` after. Because, while compose technically does take 1-to-n arguments, spreading in a set of arguments like that only works if the argument at that position is a spread and is the last argument of the function. The definition for `compose` for more that 7 arity is also incorrect (see [here](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/ramda/index.d.ts#L782)) and will be fixed in the coming months

While typescript does throw an error for `compose`, your actual code will still work
```typescript
import { compose } from 'ramda';

const foo = (a: number) => a;

const myCompose = (fns: Array<(a: number) => number>) => {
  // @ts-expect-error
  return compose(...[...fns, foo]);
};

const fn = myCompose([(a) => a * 2]);

fn(2); // 4
```

The `// @ts-expect-error` is needed here to get over the current typing limitation on `compose`, but the function itself will work

I hope that helps

**Note: Updated 3/9/23**
